test/integration/targets/ansible-test-no-tty/runme.sh

```diff
@@ -2,4 +2,6 @@
 
 set -eux
 
-./runme.py
+test ! -t 0  # isatty(stdin) is False
+test ! -t 1  # isatty(stdout) is False
+test ! -t 2  # isatty(stderr) is False
```

Comment:
```
fail() {
  echo "$@"
  exit 1
}
test ! -t 0 || fail 'stdin is not a tty'
```

Reply:
In that case, I'm going with:
```bash
unexpected=
test ! -t 0 || ${unexpected:?stdin is a tty}
test ! -t 1 || ${unexpected:?stdout is a tty}
test ! -t 2 || ${unexpected:?stderr is a tty}
```
```console
$ ./test/integration/targets/ansible-test-no-tty/runme.sh </dev/null 
+ unexpected=
+ test '!' -t 0
+ test '!' -t 1
./test/integration/targets/ansible-test-no-tty/runme.sh: line 7: unexpected: stdout is a tty
```
(And no, `test ! -t 0  ||  echo 'stdin is a tty' && exit 1` does not work because the exit is not short circuited. `test ! -t 0 || { echo 'stdin is a tty'; exit 1; }` would work.)